pkg/apis/provisioning/v1alpha5/labels.go

```diff
@@ -0,0 +1,126 @@
+/*
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha5
+
+import (
+	"fmt"
+	"strings"
+
+	v1 "k8s.io/api/core/v1"
+	stringsets "k8s.io/apimachinery/pkg/util/sets"
+)
+
+var (
+	ArchitectureAmd64    = "amd64"
+	ArchitectureArm64    = "arm64"
+	OperatingSystemLinux = "linux"
+
+	// Karpenter specific domains and labels
+	KarpenterLabelDomain = "karpenter.sh"
+	LabelCapacityType    = KarpenterLabelDomain + "/capacity-type"
+
+	// RestrictedLabelDomains are either prohibited by the kubelet or reserved by karpenter
+	RestrictedLabelDomains = stringsets.NewString(
+		"kubernetes.io",
+		"k8s.io",
+		KarpenterLabelDomain,
+	)
+
+	// LabelDomainException are sub-domains of the RestrictedLabelDomains but allowed because
+	// they are not used in a context where they may be passed as argument to kubelet.
+	LabelDomainExceptions = stringsets.NewString(
+		"kops.k8s.io",
+	)
+
+	// LabelExceptions are labels that belong to the RestrictedLabelDomains but allowed.
+	// Karpenter is aware of these labels, and they can be used to further narrow down
+	// the range of the corresponding values by either provisioner or pods.
+	LabelExceptions = stringsets.NewString(
```

Comment:
Why did you rename this from WellKnownLabels? This is a core kubernetes concept: https://kubernetes.io/docs/reference/labels-annotations-taints/

Reply:
Currently Karpenter's WellKnownLabel set is a super set of the kuberetes WellKnownLabels (e.g., karpenter.sh domain is in it).
```
	WellKnownLabels = stringsets.NewString(
		v1.LabelTopologyZone,
		v1.LabelInstanceTypeStable,
		v1.LabelArchStable,
		v1.LabelOSStable,
		LabelCapacityType,
		v1.LabelHostname, // Used internally for hostname topology spread
	)
```
It is really trying to capture the concept of restricted labels.

Reply:
Given that this is now the set of known labels (without hostname), I really think that's what we should call it.

Reply:
Agree. Done.